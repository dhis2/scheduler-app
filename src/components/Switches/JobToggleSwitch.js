import React from 'react'
import { func, string, bool } from 'prop-types'
import { connect } from 'react-redux'
import { Switch } from '@dhis2/ui-core'
import * as rootSelectors from '../../rootSelectors'
import { actions, selectors } from '../../data/jobs'

export const UnconnectedJobToggleSwitch = ({
    id,
    checked,
    enableJob,
    disableJob,
    isFetching,
}) => {
    const toggleJob = checked ? disableJob : enableJob

    return (
        <Switch
            disabled={isFetching}
            checked={checked}
            name={`toggle-${id}`}
            onChange={() => toggleJob(id)}
        />
    )
}

UnconnectedJobToggleSwitch.propTypes = {
    id: string.isRequired,
    checked: bool.isRequired,
    enableJob: func.isRequired,
    disableJob: func.isRequired,
    isFetching: bool.isRequired,
}

const mapStateToProps = state => {
    const jobs = rootSelectors.getJobs(state)

    return {
        isFetching: selectors.getIsFetching(jobs),
    }
}

const mapDispatchToProps = {
    enableJob: actions.enableJob,
    disableJob: actions.disableJob,
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(UnconnectedJobToggleSwitch)
