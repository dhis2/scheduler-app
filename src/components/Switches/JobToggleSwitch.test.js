import React from 'react'
import { shallow, mount } from 'enzyme'
import { UnconnectedJobToggleSwitch as JobToggleSwitch } from './JobToggleSwitch'

describe('<JobToggleSwitch>', () => {
    it('renders correctly', () => {
        const enableJob = () => {}
        const disableJob = () => {}
        const wrapper = shallow(
            <JobToggleSwitch
                id="1"
                checked={true}
                enableJob={enableJob}
                disableJob={disableJob}
            />
        )

        expect(wrapper).toMatchSnapshot()
    })

    it('calls enableJob when unchecked and checkbox changes to true', () => {
        const enableJob = jest.fn()
        const disableJob = () => {}
        const wrapper = mount(
            <JobToggleSwitch
                id="1"
                checked={false}
                enableJob={enableJob}
                disableJob={disableJob}
            />
        )

        wrapper.find('input').simulate('change', { target: { checked: true } })

        expect(enableJob).toHaveBeenCalled()
    })

    it('calls disableJob when checked and checkbox changes to false', () => {
        const disableJob = jest.fn()
        const enableJob = () => {}
        const wrapper = mount(
            <JobToggleSwitch
                id="1"
                checked={true}
                enableJob={enableJob}
                disableJob={disableJob}
            />
        )

        wrapper.find('input').simulate('change', { target: { checked: false } })

        expect(disableJob).toHaveBeenCalled()
    })
})
