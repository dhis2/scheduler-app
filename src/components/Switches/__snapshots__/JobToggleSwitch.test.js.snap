// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobToggleSwitch> renders correctly 1`] = `
<Switch
  checked={true}
  name="toggle-1"
  onChange={[Function]}
/>
`;
